src/docs/reference/core/Configuration/README.md

```diff
@@ -18,6 +18,9 @@ Orchard Core supports a hierarchy of Configuration Sources
 
 The Configuration Sources are loaded in the above order, and settings lower in the hierarchy will override values configured higher up, i.e. an Global Tenant value will always be overridden by an Environment Variable.
 
+!!! note 
+    Modules can expose configuration via `IOptions` objects, `IShellConfiguration` keys (like the ones supplied in `appsettings.json` files) or both. However, one is not guarantee for the other. For example, the Email module allows SMTP configuration via the `SmtpSettings` class exposed via `IOptions` but to supply those settings from an `appsettings.json` file you'll need read out `IShellConfiguration` keys in your own app's code, see below.
```

Comment:
Cool, nice move on the node_modules.

I can't edit this :(

I think editing your prs might be disabled because you are pushing prs from an organisation repo?
That may be a new setting on github, or a github bug, as I used to push editable prs from my organisations repo and @jtkech could edit them. But recently @Skrypt tried to, and could not without me adding him to the repo's security.

Makes it difficult to review .

This sentence doesn't read correctly, nor explain / show how to do what you are suggesting.
Nor will the example work, because on the host builder you cannot inject `IShellConfiguration`

Suggest moving the example to the Email docs, making it one that works, include a link here, and make the note make sense. (if I could just edit it I would)

Reply:
I don't see any option here to change accesses here, so I don't know, it's strange. However, you should still be able "Insert a suggestion" in comments.

You can resolve `IShellConfiguration` from the builder though like here: https://github.com/OrchardCMS/OrchardCore/blob/2b961369fa9401702e1d7b420d3281b0f71e269b/src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs#L192

How to use `IOptions` is explained below, as is `IShellConfiguration`, and what's not related to it is in the linked docs (as it's not specific to Orchard). So I wouldn't reiterate those here. The point of this note is to prevent people from thinking "cool, so there is static configuration for everything, I'll just follow the pattern" as it happened here: https://github.com/OrchardCMS/OrchardCore/issues/6264

I don't really see how this doesn't read correctly, please explain.

Reply:
Ok, so it's difficult to just make a suggestion here, because so may things...

The overload you referenced is for the `Configure` action. 
Here you are working with the `ConfigureServices` action.
When you're in a `ConfigureServices` action you can't resolve a service, because you are in the process of registering services.

So there are a number of ways to do what you are suggesting, but they are not straight forward.

At the web host `Startup` you could resolve/build `IConfiguration`, not `IShellConfiguration` because it is not registered.
Then you can key down into the `OrchardCore` sections, but you do not have to, the keys could be stored outside of that section.

You could also resolve `IShellConfiguration` from `ShellScope.Services` inside the lambda.

But that is a funny pattern to be putting in the docs.

So again what I would recommend is putting an actual working example in the email docs. Perhaps from a module `Startup.cs` where you can easily inject `IShellConfiguration` in the `Startup` constructor.

The difficulty with the sentence is it is really ambigious and after reading it three times I still don't know what it says.

So that's why I am suggesting you do an example that works, specific for email, and link to that.

Reply:
OK, but we still need a warning here. The point is to prevent people from getting confused like it happened under the linked issue. Having an example of the code is a next step for those reading this.

What would you write as a warning, assuming there is an example to link to? I have no trouble understanding what I've written :) and I don't know how it can be easier to understand while not omitting what needs to be conveyed.

Reply:
ok, so we're both wrong. JT is way ahead of me, and there is another overload with a service provider. So here's a working example
```
            services.AddOrchardCms()
                .ConfigureServices((tenantServices, serviceProvider) =>
                {
                    var shellConfiguration = serviceProvider.GetRequiredService<IShellConfiguration>();
                    var password = shellConfiguration.GetValue("SmtpSettings:Password", "empty");
                    tenantServices.PostConfigure<SmtpSettings>(o =>
                    {
                        o.Password = password;
                        o.Port = 255;
                    });
                });
```

So I would just suggest to use the example and don't have a warning at all.

Reply:
Thanks, I've updated the example with this too. I don't see why a warning at the beginning of this section is not helpful in preventing people from jumping to conclusions which we see happening. I've made it shorter though.